clicache/src/impl/DelegateWrapper.hpp

```diff
@@ -80,16 +80,18 @@ namespace Apache
         /// Native <c>apache::geode::client::Serializable</c> object after invoking the managed
         /// delegate and wrapping inside a <c>ManagedCacheableKey</c> object.
         /// </returns>
-        apache::geode::client::Serializable* NativeDelegateGeneric( )
+        std::shared_ptr<apache::geode::client::Serializable> NativeDelegateGeneric( )
         {
           auto tempObj = m_delegate( );
           if(auto tempDelta = dynamic_cast<IGeodeDelta^>(tempObj))
           {
-            return new apache::geode::client::ManagedCacheableDeltaGeneric(tempDelta);
+            return std::shared_ptr<apache::geode::client::ManagedCacheableDeltaGeneric>(
```

Comment:
Did `std::make_shared` not work here?

Reply:
No, common problem we ran into in C++/CLR. If a parameter to the constructor is a GC reference or pointer the `std::forward` in the `std::shared_ptr` constructor fails.